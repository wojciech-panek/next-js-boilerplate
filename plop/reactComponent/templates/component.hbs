import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { compose } from 'ramda';

import withIntl from '../../shared/lib/withIntl'; // TODO fix relative import
import { Container } from './{{ camelCase name }}.styles';


export class {{ pascalCase name }} extends PureComponent {
  static propTypes = {
  };

  render() {
    return (
      <Container>
        {{ pascalCase name }} component
      </Container>
    );
  }
}

export default compose(
  withIntl,
)({{ pascalCase name }});

