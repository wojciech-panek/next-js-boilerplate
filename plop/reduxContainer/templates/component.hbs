import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';


export class {{ pascalCase name }} extends PureComponent {
  static propTypes = {
  };

  render() {
    return (
      <div>
        {{ pascalCase name }} component
      </div>
    );
  }
}
