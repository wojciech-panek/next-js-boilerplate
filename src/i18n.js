const addLocaleData = require('react-intl').addLocaleData;
const enLocaleData = require('react-intl/locale-data/en');
const plLocaleData = require('react-intl/locale-data/pl');

const enTranslationMessages = require('./translations/en.json');
const plTranslationMessages = require('./translations/pl.json');

const LOCALES = {
  ENGLISH: 'en',
  POLISH: 'pl',
};

const DEFAULT_LOCALE = LOCALES.ENGLISH;

const appLocales = [
  LOCALES.ENGLISH,
  LOCALES.POLISH,
];

const formatTranslationMessages = (locale, messages) => {
  const defaultFormattedMessages = locale !== DEFAULT_LOCALE
    ? formatTranslationMessages(DEFAULT_LOCALE, enTranslationMessages)
    : {};
  return Object.keys(messages).reduce((formattedMessages, key) => {
    const formattedMessage = !messages[key] && locale !== DEFAULT_LOCALE
      ? defaultFormattedMessages[key]
      : messages[key];
    return Object.assign(formattedMessages, { [key]: formattedMessage });
  }, {});
};

const translationMessages = {
  [LOCALES.ENGLISH]: formatTranslationMessages(LOCALES.ENGLISH, enTranslationMessages),
  [LOCALES.POLISH]: formatTranslationMessages(LOCALES.POLISH, plTranslationMessages),
};

if (typeof window !== 'undefined') {
  addLocaleData(enLocaleData);
  addLocaleData(plLocaleData);

  if (!window.Intl) {
    (new Promise((resolve) => {
      resolve(require('intl'));
    }))
      .then(() => Promise.all([
        require('intl/locale-data/jsonp/en.js'),
        require('intl/locale-data/jsonp/pl.js'),
      ]))
      .catch((err) => {
        throw err;
      });
  }
}

module.exports = {
  LOCALES,
  DEFAULT_LOCALE,
  appLocales,
  translationMessages,
};
