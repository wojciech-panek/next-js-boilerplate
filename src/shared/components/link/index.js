export { default as Link } from './link.container';
