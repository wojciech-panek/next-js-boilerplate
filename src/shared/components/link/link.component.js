import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

import { Link as NextLink } from '../../../routes';
import { DEFAULT_LOCALE } from '../../../i18n';


export class Link extends PureComponent {
  static propTypes = {
    to: PropTypes.string.isRequired,
    language: PropTypes.string.isRequired,
    children: PropTypes.node.isRequired,
    intl: PropTypes.object.isRequired,
    router: PropTypes.object.isRequired,
  };


  render() {
    const { to, language, intl, router, ...restOfProps } = this.props; //eslint-disable-line no-unused-vars

    let toWithLocale = to;
    if (language !== DEFAULT_LOCALE) {
      toWithLocale = `/${language}${toWithLocale}`;
    }

    return (
      <NextLink {...restOfProps} to={toWithLocale}>{this.props.children}</NextLink>
    );
  }
}
