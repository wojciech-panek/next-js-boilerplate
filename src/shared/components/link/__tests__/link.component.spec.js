import React from 'react';
import { shallow } from 'enzyme';

import { Link } from '../link.component';
import { DEFAULT_LOCALE } from '../../../../i18n';


describe('Link: Component', () => {
  const defaultProps = {
    intl: {},
    router: {},
  };

  const component = (props) => (
    <Link {...defaultProps} {...props}>link</Link>
  );

  it('should render correctly for DEFAULT_LANGUAGE link', () => {
    const wrapper = shallow(component({ language: DEFAULT_LOCALE, to: '/example/nested/url' }));
    global.expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly for non DEFAULT_LANGUAGE link', () => {
    const wrapper = shallow(component({ language: 'language', to: '/example/nested/url' }));
    global.expect(wrapper).toMatchSnapshot();
  });
});
