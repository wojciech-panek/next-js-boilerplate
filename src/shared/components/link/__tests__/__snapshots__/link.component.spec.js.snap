// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link: Component should render correctly for DEFAULT_LANGUAGE link 1`] = `
<LinkRoutes
  to="/example/nested/url"
>
  link
</LinkRoutes>
`;

exports[`Link: Component should render correctly for non DEFAULT_LANGUAGE link 1`] = `
<LinkRoutes
  to="/language/example/nested/url"
>
  link
</LinkRoutes>
`;
